test/gtest/ucs/test_hash_perf.cc

```diff
@@ -0,0 +1,281 @@
+        khiter_t ep_found = kh_get(khash_ep_hash, obj, key);
+        EXPECT_TRUE(ep_found != kh_end(obj));
+        return kh_value(obj, ep_found);
+    }
+
+    void del_elems() {
+        for (khiter_t it = kh_begin(obj); it != kh_end(obj); ++it) {
+            if (!kh_exist(obj, it)) {
+                continue;
+            }
+            Hash_type *ep = kh_value(obj, it);
+            free(ep);
+        }
+        kh_destroy(khash_ep_hash, obj);
+    }
+
+    khash_t(khash_ep_hash) *obj;
+};
+
+#define SGLIB_HASH_SIZE                 32767
+#define test_sglib_compare(_ep1, _ep2)  ((int64_t)(_ep1)->dest_uuid - (int64_t)(_ep2)->dest_uuid)
+#define test_sglib_hash(_ep)            ((_ep)->dest_uuid)
+
+struct SGLIB_hash_type : public Hash_type {
+    SGLIB_hash_type *next;
+};
+
+SGLIB_DEFINE_LIST_PROTOTYPES(SGLIB_hash_type, test_sglib_compare, next);
+SGLIB_DEFINE_LIST_FUNCTIONS(SGLIB_hash_type, test_sglib_compare, next);
+SGLIB_DEFINE_HASHED_CONTAINER_PROTOTYPES(SGLIB_hash_type, SGLIB_HASH_SIZE, test_sglib_hash);
+SGLIB_DEFINE_HASHED_CONTAINER_FUNCTIONS(SGLIB_hash_type, SGLIB_HASH_SIZE, test_sglib_hash);
+
+struct Perf_compare_sglib : public Perf_compare_base {
+
+    const char *get_name() const {
+        return "sglib";
+    }
+
+    void init_storage() {
+        obj = (SGLIB_hash_type **) ucs_malloc(sizeof(*obj) * SGLIB_HASH_SIZE, "sglib_hash");
+        sglib_hashed_SGLIB_hash_type_init(obj);
+    }
+
+    void add_elem(const uint64_t key) {
+        SGLIB_hash_type *ep = create_element<SGLIB_hash_type>(key);
+        sglib_hashed_SGLIB_hash_type_add(obj, ep);
+    }
+
+    Hash_type *find_elem(const uint64_t key) const {
+        SGLIB_hash_type ep, *ep_found = 0;
+        ep.dest_uuid = key;
+        ep_found = sglib_hashed_SGLIB_hash_type_find_member(obj, &ep);
+        EXPECT_TRUE(ep_found);
+        return ep_found;
+    }
+
+    void del_elems() {
+        struct sglib_hashed_SGLIB_hash_type_iterator it;
+        for (SGLIB_hash_type *local_it = sglib_hashed_SGLIB_hash_type_it_init(&it, obj); local_it != NULL; local_it = sglib_hashed_SGLIB_hash_type_it_next(&it)) {
+            sglib_hashed_SGLIB_hash_type_delete(obj, local_it);
+            free(local_it);
+        }
+        free(obj);
+    }
+
+    SGLIB_hash_type **obj;
+};
+
+class test_hash_perf : public ucs::test {
+};
+
+UCS_TEST_F(test_hash_perf, perf_compare) {
+
+    size_t trip_counts[] = {1, 2, 8, 128, 1024, 32768, 262144, 1048576, 0};
+    ucs_time_t khash_lookup_ns_1024 = 0;
+    ucs_time_t sglib_lookup_ns_1024 = 0;
+    Perf_compare_base *perf_compare_khash_ptr = new Perf_compare_khash;
+    Perf_compare_base *perf_compare_sglib_ptr = new Perf_compare_sglib;
+    Perf_compare_base *hashes[] = {
+        perf_compare_khash_ptr,
+        perf_compare_sglib_ptr,
+        new Perf_compare_map,
+        new Perf_compare_unordered_map,
+        NULL
+    };
+
+    if (ucs::test_time_multiplier() > 1) {
+        UCS_TEST_SKIP;
+    }
```

Comment:
May be it is better to allow running this test under valgrind but avoid performance comparison.